docs: Improvements to mobile instructions [skip ci]

schjonhaug opened the pull request:
Improvements on instructions on how to build for mobile:
* Cleaned up markdown headings
* Included necessary commands to install and init gomobile
* Added command on how to go to the correct src folder before building libraries (due to a known limitation in gomobile)
* Added command on how to checkout the last tagged release of lnd before building libraries
* Fixed incorrect link to Dockerfile
* Use go 1.17.3 instead of 1.17.2 in the Dockerfile.
* For convenience, added command on how to move all the generated Swift protoc definition files to the  same directory as the framework file.

guggero reviewed (commented):
Thanks for the update, appreciate you helping out others with build problems and updating the docs at the same time :100: 

Just a few mostly non-blocking comments.

schjonhaug commented on docs/release-notes/release-notes-0.15.0.md:
```diff
@@ -14,16 +14,19 @@
 * [An example systemd service file](https://github.com/lightningnetwork/lnd/pull/6033)
   for running lnd alongside a bitcoind service is now provided in
-  `contrib/init/lnd.service`.
+`contrib/init/lnd.service` .
```
It’s the formatter I’m using, sorry about that. Will revert it. Also, should I add my changes to 0.15.0 or perhaps to 0.14.2 instead?

guggero replied:
No problem! I think 0.15 is okay for this. It'll be looked at on the `master` branch mostly anyway, so better to aim for the less timing sensitive 0.15 branch than to try to make it into 0.14.2 and then needing to rebase again if that window is missed.

schjonhaug commented on mobile/README.md:
```diff
@@ -1,66 +1,89 @@
 After the build has succeeded, the libraries will be found in
 `mobile/build/ios/Lndmobile.framework` and
-`mobile/build/android/Lndmobile.aar`. Reference your platforms' SDK
+`mobile/build/android/Lndmobile.aar` . Reference your platforms' SDK
```
Yes, it’s the formatter. Will revert.

schjonhaug commented on docs/release-notes/release-notes-0.15.0.md:
```diff
@@ -14,16 +14,19 @@
-
 # Contributors (Alphabetical Order)
-
+* Andreas Schjønhaug
```
Sorry, it’s the formatter. Will revert.

schjonhaug commented on mobile/README.md:
````diff
@@ -1,66 +1,89 @@
+Install [falafel](https://github.com/lightninglabs/falafel):
+
+```shell
+⛰  go get -u -v github.com/lightninglabs/falafel
````
That’s a good point, fixed.

schjonhaug commented:
> Thanks for the update, appreciate you helping out others with build problems and updating the docs at the same time 💯
> 
> Just a few mostly non-blocking comments.

Thanks for your feedback, I believe to have fixed them all now.

guggero commented on docs/release-notes/release-notes-0.15.0.md:
```diff
@@ -16,6 +16,10 @@
 
+## Mobile
+
+* Improved instructions on [how to build lnd for mobile](https://github.com/lightningnetwork/lnd/tree/master/mobile).
```
The release notes should also include a link to the PR itself, for easy reference of the actual change.

guggero reviewed (commented):
Thanks for the quick update. Please squash your commits and add a link to the PR itself to the release notes. Then we're good to go IMO :tada:

schjonhaug commented:
> Thanks for the quick update. Please squash your commits and add a link to the PR itself to the release notes. Then we're good to go IMO 🎉

No problem. Fixed and squashed.

carlaKC commented on docs/release-notes/release-notes-0.15.0.md:
```diff
@@ -26,6 +26,10 @@
   `pending_force_closing_channels` under `pendingchannels` whereas before was
   empty(zero).
 
+## Documentation
```
nit: in commit message `Swift Protobug` -> `Swift Protobuf`?

schjonhaug replied:
Thanks, fixed.

carlaKC reviewed (approved):
Thanks for the updates! Just one typo nit in the commit message, non-blocking
